types/ramda/index.d.ts

```diff
@@ -1388,6 +1399,32 @@ export function minBy<T>(keyFn: (a: T) => Ord, a: T, b: T): T;
 export function minBy<T>(keyFn: (a: T) => Ord, a: T): (b: T) => T;
 export function minBy<T>(keyFn: (a: T) => Ord): _.F.Curry<(a: T, b: T) => T>;
 
+/**
+ * Creates a copy of the passed object by applying an fn function to the given prop property.
+ * The function will not be invoked, and the object will not change
+ * if its corresponding property does not exist in the object.
+ * All non-primitive properties are copied to the new object by reference.
+ */
+export function modify<T extends Record<K, V>, K extends string | number | symbol, F extends (a: V) => any, V>(prop: K, fn: F, object: T): Omit<T, K> & Record<K, ReturnType<F>>;
+export function modify<K extends string | number | symbol, F extends (a: V) => any, V>(prop: K, fn: F): <T extends Record<K, V>>(object: T) => Omit<T, K> & Record<K, ReturnType<F>>;
+export function modify<K extends string | number | symbol>(prop: K): <T extends Record<K, V>, F extends (a: V) => any, V>(fn: F, object: T) => Omit<T, K> & Record<K, ReturnType<F>>;
```

Comment:
This could further be improved like this, so we end up with a simpler type in trivial cases.

```typescript
export function modify<T extends Record<K, any>, K extends string | number | symbol, R>(prop: K, fn: (a: T[K]) => R, object: T): R extends T[K] ? T : Omit<T, K> & Record<K, R>;
```

Reply:
Or with additional overloads like this:
```
function modify2<T extends Record<K, any>, K extends string | number | symbol>(prop: K, fn: (a: T[K]) => T[K], object: T): T;
function modify2<T extends Record<K, any>, K extends string | number | symbol, R>(prop: K, fn: (a: T[K]) => R, object: T): Omit<T, K> & Record<K, R>;
```